packages/client/src/components/app/Modal.svelte

```diff
@@ -0,0 +1,147 @@
+<script>
+  import { fade, fly } from "svelte/transition"
+  import { getContext } from "svelte"
+  import { clickOutside, Icon } from "@budibase/bbui"
+  import Portal from "svelte-portal"
+
+  const component = getContext("component")
+  const { styleable, modalStore, builderStore, dndIsDragging } =
+    getContext("sdk")
+
+  export let onClose = async () => {}
+  export let ignoreClicksOutside
+  export let size
+
+  // Open modal automatically in builder
+  $: {
+    if ($builderStore.inBuilder) {
+      if (
+        $component.inSelectedPath &&
+        $modalStore.contentId !== $component.id
+      ) {
+        modalStore.actions.open($component.id)
+      } else if (
+        !$component.inSelectedPath &&
+        $modalStore.contentId === $component.id &&
+        !$dndIsDragging
+      ) {
+        modalStore.actions.close()
+      }
+    }
+  }
+
+  $: open = $modalStore.contentId === $component.id
+
+  const handleModalClose = async () => {
+    modalStore.actions.close()
+    await onClose()
```

Comment:
Was a little confused by this at first, as I thought my default argument for the prop of an empty async function would be sufficient. But turns out if you add an on-close action, then remove it, you get the error you mentioned.

Maybe my svelte-fu is lacking, because I'd assume the parent of this component is passing something nullish to this prop, which should then use the default 🤷 Maybe it's a bound prop or something, idk, but good catch, I'll add the guard.

Reply:
Svelte default props work the same as default function params - they only get used if the value is `undefined`, but not for `null`. Just to explain why that's probably happening.